### ISSUE TITLE
Missing Permissions in catalog management roles

### ISSUE THREAD

**Issue by ryansann** [rancher/rancher#34592]
**Rancher Server Setup**
- Rancher version: v2.6.0
- Installation option (Docker install/Helm Chart): Docker install

**Information about the Cluster**
- Cluster Type (Local/Downstream): local

**Describe the bug**

The builtin `catlaogs-manage` GlobalRole is missing permissions for verb: `*` on resource: `clusterrepos`, this permissions grants the ability to manage chart repositories. 

**To Reproduce**
 
- After Rancher starts, run: `kubectl get globalroles catalogs-manage -o yaml` against the local cluster

**Result**
```yaml
rules:
- apiGroups:
  - management.cattle.io
  resources:
  - catalogs
  - templates
  - templateversions
  verbs:
  - '*'
```

**Expected Result**

```yaml
rules:
- apiGroups:
  - management.cattle.io
  resources:
  - catalogs
  - templates
  - templateversions
  verbs:
  - '*'
- apiGroups:
  - catalog.cattle.io
  resources:
  - clusterrepos
  verbs:
  - '*'
```

**Comment by ryansann** [comment]
Testing steps:

Start Rancher v2.6.0 (version without this change)
Create a standard user, check the "Configure Catalogs" builtin role
Add new user as a member to a cluster (local or downstream)
Log in as user and go to Apps & Marketplace > Chart Repositories in the cluster
Note that chart repositories cannot be modified in any way
Upgrade to this PR
Refresh browser, user can modify chart repositories
catalogs-manage GlobalRole:

rules:
- apiGroups:
  - management.cattle.io
  resources:
  - catalogs
  - templates
  - templateversions
  verbs:
  - '*'
- apiGroups:
  - catalog.cattle.io
  resources:
  - clusterrepos
  verbs:
  - '*'

**Comment by ryansann** [comment]
This issue needs to be backported to 2.5

**Comment by ryansann** [comment]
Small change to this. After some discussions, I've also added:
```yaml
- apiGroups:
  - catalog.cattle.io
  resources:
  - clusterrepos
  verbs:
  - '*'
```
to the `clustercatalogs-manage` Role.

The `clusterrepo` permissions are not being added to the `projectcatalogs-manage` role because there is no concept of project level catalogs in dashboard / `catalog.cattle.io`

**Comment by anupama2501** [comment]
Reproduced on v2.6.0

1. Created an rke1 and rke2 node driver clusters
2. Created a standard user -user1, and added `configure catalog` option in the global permissions.
3. Added the user as a cluster member and logged in as user1
4. From apps & market --> chart repos no options were available for the user to change the repos

**Comment by anupama2501** [comment]
On v2.6-head 1a57856

Api groups are added in the rules in the local cluster after upgrading the rancher server:
`kubectl get globalroles catalogs-manage -o yaml`

```
rules:
- apiGroups:
  - management.cattle.io
  resources:
  - catalogs
  - templates
  - templateversions
  verbs:
  - '*'
- apiGroups:
  - catalog.cattle.io
  resources:
  - clusterrepos
  verbs:
  - '*'
```

- Upgraded rancher server to v2.6-head
- Created a new user with `Configure cluster` in the global permissions assigned.
- Added the user in the local, rke1 and rke2 cluster as a cluster member
- standard user in the local cluster, can edit/delete/create repositories from apps&Market --> charts
- User in rke1 and rke2 downstream clusters for the same user cannot still edit/remove/add

@ryansann 
Standard user in rke1 and rke2 clusters can edit the repositories if we add an additional permission to the user in the cluster RBAC `Manage Cluster Catalogs` but for local cluster even without that permission a cluster member can manage the catalogs.

**Comment by cbron** [comment]
Confused at that behavior, they are technically the same thing, the only difference is that one is a global permission and one is cluster-level. Once you get that global permissions it should work in downstream clusters, but perhaps . @jiaqiluo can you take a look.
@dramich is it because global context role's don't get synced downstream ? I looked and most of those are mgmt-cluster specific like settings and features. If that is the case then I suppose this is working correctly, and we should remove it from the global role to avoid confusion. Tell users they must use "manage cluster catalogs" even for local cluster, since those aren't really global anymore.

**Comment by cbron** [comment]
@jiaqiluo I have moved this to 2.6.2 so we can discuss proper approach, can you revert anything already merged.

**Comment by brudnak** [comment]
## Unsuccessful Validation ❌ 

| Component               | Version / Type                                     |
| ----------------------- | -------------------------------------------------- |
| Rancher version         | v2.6-6b00d71d60667a5fa878ddc3e2dc9df9e5bd78a8-head |
| Rancher cluster type    | Single Node Docker Install                         |
| Docker version          | 20.10.7, build 20.10.7-0ubuntu5~20.04.2            |
| Helm version            | v2.16.8-rancher1                                   |
| Downstream cluster type | RKE1 Linode                                        |
| Downstream K8s version  | v1.21.6-rancher1-1                                 |
| Browser type            | Google Chrome                                      |
| Browser version         | 96.0.4664.45 (Official Build) (x86_64)             |


##  steps

1. After creating a downstream cluster as described in the table above. Followed these steps
2. Starting from the default Rancher homepage `/dashboard/home`
3. Click the `hamburger menu` -> click `Users & Authentication` -> click `Create`
4. Fill out and submit the form to create a new user
    - Username: `user1`
    - Display Name: `user1`
    - Enter a password for `user1`
    - Global Permissions: select -> `Standard User`
    - Built-in: select -> `Configure Catalogs`
    - Click `Create` -> you are redirected back to the users page, you will see the newly created `user1` and the `Default Admin`
5. Click the `hamburger menu` -> under the explore cluster section in the menu click `local` -> then from the same side menu click `RBAC` then from `Cluster Members` click -> `Add`
6. Search for `user1` and for Cluster Permissions select `Member` -> then click `Create`
7. Now click the user icon on the top right of the screen and click `Log Out`
8. Now sign in as `user1` with the password you provided during the creation of `user1`
9. Now as `user1` click the `hamburger menu` -> click `local` -> click `Apps & Marketplace` -> click `Repositories` click the `ellipsis (three dots)` next to all three available repositories and notice there is no edit or delete options available. See screen shot labeled `A1`

![Screen Shot 2021-11-16 at 12 47 40 PM](https://user-images.githubusercontent.com/40877149/142059585-ac5055e4-475a-45d7-9d94-b0755805d0c5.png)

10. Log out of `user1`
11. Log back in as `admin`
12. Add `user1` as a `Member` of the downstream cluster described in the setup table above
13. Log out of `admin` and back into `user1`
14. As `user1` click the `hamburger menu` -> under the explore cluster section in the menu select the downstream cluster you create -> click `Apps & Marketplace` -> `Repositories`
15. Click the `ellipsis (three dots)` next to the repositories and see there is no edit or delete options


## Additional Info

When logged in as `admin` if you follow these steps:
1. Click `hamburger menu` -> click `Users & Authentication` -> click `Roles` ->  click `Configure Catalogs`
2. You can see that configure catalogs has `Create Delete Get List Patch Update Watch` for resources `catalogs,templates,templateversions`
3. See screen shot below labeled `A2`

![Screen Shot 2021-11-16 at 1 01 34 PM](https://user-images.githubusercontent.com/40877149/142059535-bced73d0-096f-479d-9445-93050c810550.png)

## Video

https://user-images.githubusercontent.com/40877149/142059654-fe49239f-bbf9-476a-9a1f-a7a7adb38daf.mov

## Workaround

There appears to be a workaround using the old UI.

1. Be logged in as `user1`
2. In the URL start from `/g`
3. Under Clusters click the `local` cluster
4. Under the `Tools` dropdown select `Catalogs`
5. When clicking the `ellipsis (three dots)` next to the catalogs/"repositories" in the old UI. You can see `edit` or `delete` options available
6. Selecting edit or delete from these has working functionality

**Comment by ryansann** [comment]
@brudnak and I discussed further how to validate this. The user needs to be associated to the role when they are added to a cluster, not during creation. 

The `Manage Catalogs` Global Role, seen during user creation, is not properly changing its display name to `Legacy Manage Catalogs` per the code changes. This likely has to do with dashboard display name logic, I will figure that part out and get this back into test soon.

**Comment by brendarearden** [comment]
Available to test in v2.6.3-rc2

**Comment by brudnak** [comment]
# My checks PASSED

## Reproduction Environment

| Component               | Version / Type                          |
| ----------------------- | --------------------------------------- |
| Rancher version         | v2.6.0                                  |
| Rancher cluster type    | Docker                                  |
| Docker version          | 20.10.7, build 20.10.7-0ubuntu5~20.04.2 |
| Helm version            | v2.16.8-rancher1                        |
| Downstream cluster type | RKE1 Linode                             |
| Downstream K8s version  | v1.21.6-rancher1-1                      |
| Browser type            | Google Chrome                           |
| Browser version         | 96.0.4664.55 (Official Build) (x86_64)  |

## Reproduction steps

1. Starting from the **default** Rancher homepage `/dashboard/home`
2. Click `hamburger menu` -> click `Users & Authentication` -> click `Create`
3. Fill out the user creation form
    - Username: `user1`
    - Password: *`set some password`*
    - Global Permissions: `Standard User`
    - Click `Create`
4. Click `hamburger menu` -> click the downstream cluster under `EXPLORE CLUSTER` -> click `RBAC` -> in `Clust Memebers` click `Add`
5. Fill out the form
   - Select Member: search for `user1`
   -  Click `Custom` -> select `Manage Cluster Catalogs`
   -  Click `Create`
6. Also add a standard member permission for `user1` 
7. Click the user icon at the top right -> click `Log Out`
8. Sign in as `user1`
9. Click `hamburger menu` -> click assigned cluster under `EXPLORE CLUSTER` -> click `Apps & Marketplace` -> click `Chart Repositories` -> click the `ellipsis (three dots)` next to the repositories

**RESULT**

There is no option to delete or edit when clicking the ellipsis for the repositories.

<hr>
  
## Validation Environment

| Component               | Version / Type                          |
| ----------------------- | --------------------------------------- |
| Rancher version         | v2.6.3-rc2                              |
| Rancher cluster type    | Docker                                  |
| Docker version          | 20.10.7, build 20.10.7-0ubuntu5~20.04.2 |
| Helm version            | v2.16.8-rancher1                        |
| Downstream cluster type | RKE1 Linode                             |
| Downstream K8s version  | v1.21.6-rancher1-1                      |
| Browser type            | Google Chrome                           |
| Browser version         | 96.0.4664.55 (Official Build) (x86_64)  |

## Validation steps

1. Starting from the **default** Rancher homepage `/dashboard/home`
2. Click `hamburger menu` -> click `Users & Authentication` -> click `Create`
3. Fill out the user creation form
    - Username: `user1`
    - Password: *`set some password`*
    - Global Permissions: `Standard User`
    - Click `Create`
4. Click `hamburger menu` -> click the downstream cluster under `EXPLORE CLUSTER` -> click `RBAC` -> in `Clust Memebers` click `Add`
5. Fill out the form
   - Select Member: search for `user1`
   -  Click `Custom` -> select `Manage Cluster Catalogs`
   -  Click `Create`
6. Also add a standard member permission for `user1` 
7. Click the user icon at the top right -> click `Log Out`
8. Sign in as `user1`
9. Click `hamburger menu` -> click assigned cluster under `EXPLORE CLUSTER` -> click `Apps & Marketplace` -> click `Chart Repositories` -> click the `ellipsis (three dots)` next to the repositories

**RESULT**

The user is able to edit / delete repositories listed now.
